src/sidebar/store/modules/test/drafts-test.js

```diff
@@ -0,0 +1,228 @@
+    });
+  });
+
+  describe('#createDraft', () => {
+    it('should save changes', () => {
+      assert.notOk(store.getDraft(fixtures.annotation));
+      store.createDraft(fixtures.annotation, fixtures.draftWithText);
+      assert.deepEqual(
+        store.getDraft(fixtures.annotation),
+        new Draft(fixtures.annotation, fixtures.draftWithText)
+      );
+    });
+
+    it('should replace existing drafts with the same ID', () => {
+      const fakeAnnotation = {
+        id: 'my_annotation',
+      };
+      const fakeDraft = {
+        isPrivate: true,
+        tags: ['foo'],
+        text: '',
+      };
+      store.createDraft(fakeAnnotation, {
+        ...fakeDraft,
+        text: 'foo',
+      });
+      assert.equal(store.getDraft(fakeAnnotation).text, 'foo');
+
+      // now replace the draft
+      store.createDraft(fakeAnnotation, {
+        ...fakeDraft,
+        text: 'bar',
+      });
+      assert.equal(store.getDraft(fakeAnnotation).text, 'bar');
+    });
+
+    it('should replace existing drafts with the same tag', () => {
+      const fakeAnnotation = {
+        $tag: 'my_annotation_tag',
+      };
+      const fakeDraft = {
+        isPrivate: true,
+        tags: ['foo'],
+        text: '',
+      };
+      store.createDraft(fakeAnnotation, {
+        ...fakeDraft,
+        text: 'foo',
+      });
+      assert.equal(store.getDraft(fakeAnnotation).text, 'foo');
+
+      // now replace the draft
+      store.createDraft(fakeAnnotation, {
+        ...fakeDraft,
+        text: 'bar',
+      });
+      assert.equal(store.getDraft(fakeAnnotation).text, 'bar');
+    });
+  });
+
+  describe('#countDrafts', () => {
+    it('should count drafts', () => {
+      assert.equal(store.countDrafts(), 0);
+      store.createDraft({ id: '1' }, fixtures.draftWithText);
+      assert.equal(store.countDrafts(), 1);
+
+      // since same id, this  performs a replace, should still be 1 count
+      store.createDraft({ id: '1' }, fixtures.draftWithText);
+      assert.equal(store.countDrafts(), 1);
+
+      store.createDraft({ id: '2' }, fixtures.draftWithText);
+      assert.equal(store.countDrafts(), 2);
+    });
+  });
+
+  describe('#discardAllDrafts', () => {
+    it('should remove all drafts', () => {
+      store.createDraft({ id: '1' }, fixtures.draftWithText);
+      store.createDraft({ id: '2' }, fixtures.draftWithText);
+      store.discardAllDrafts(fixtures.annotation);
+      assert.equal(store.countDrafts(), 0);
+    });
+  });
+
+  describe('#removeDraft', () => {
+    it('should remove drafts', () => {
+      store.createDraft(fixtures.annotation, fixtures.draftWithText);
+      assert.isOk(store.getDraft(fixtures.annotation));
+      store.removeDraft(fixtures.annotation);
+      assert.isNotOk(store.getDraft(fixtures.annotation));
+    });
+  });
+
+  describe('#unsavedAnnotations', () => {
+    it('should return drafts for unsaved annotations which have no ID', () => {
```

Comment:
```suggestion
    it('should return unsaved annotations which have drafts', () => {
```

And a similar adjustment in the test below. The previous description implied that the return value was a `Draft`.